kernel/src/context.rs

```diff
@@ -13,21 +13,21 @@ use ostd::{
 use crate::{
     prelude::*,
     process::{
-        posix_thread::{PosixThread, ThreadLocal},
-        Process,
+        posix_thread::{CurrentPosixThread, ThreadLocal},
+        CurrentProcess,
     },
-    thread::Thread,
+    thread::CurrentThreadRef,
     util::{MultiRead, VmReaderArray},
     vm::vmar::Vmar,
 };
 
 /// The context that can be accessed from the current POSIX thread.
 #[derive(Clone)]
 pub struct Context<'a> {
-    pub process: &'a Process,
+    pub process: &'a CurrentProcess,
     pub thread_local: &'a ThreadLocal,
-    pub posix_thread: &'a PosixThread,
-    pub thread: &'a Thread,
+    pub posix_thread: &'a CurrentPosixThread<'a>,
+    pub thread: &'a CurrentThreadRef<'a>,
```

Comment:
Try to avoid pointer to pointer in this structure. I think we can have:
```rust
    pub posix_thread: CurrentPosixThread<'a>,
    pub thread: CurrentThreadRef<'a>,
```

Also, why does `CurrentThreadRef` end with "Ref" but `CurrentPosixThread` does not?